extensions/amp-story/1.0/utils.js

```diff
@@ -274,7 +280,7 @@ export function setHistoryState(win, stateName, value) {
     [stateName]: value,
   };
 
-  history.replaceState(newHistory, '');
+  throttleReplace(history, newHistory);
```

Comment:
This LGTM, but I really feel strongly about the issue number (2).
All it takes to make it work is having a `let historyStateBuffer = getState(history);` that is then updated on each `setHistoryState`, and returned on each `getHistoryState`. This `historyStateBuffer` is only pushed into the actual historyState every X ms.

(Given that we can do 100 replaceStates per 30sec, maybe we can lower the throttle limit?)

Reply:
Done.